app/models/user.rb

```diff
@@ -3,11 +3,29 @@ class User < ApplicationRecord
 
   has_many :posts, dependent: :destroy
 
+  has_many :active_relationships, class_name: 'Relationship', foreign_key: 'following_id', dependent: :destroy
+  has_many :passive_relationships, class_name: 'Relationship', foreign_key: 'followed_id', dependent: :destroy
+
+  has_many :followings, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :following
+
   validates :name, presence: true, length: { maximum: 20 },
                    format: { with: /^[0-9a-zA-Z]+$/, message: I18n.t('activerecord.errors.user.name_format_error'), multiline: true }, uniqueness: true
   validates :introduction, length: { maximum: 200 }
   validates :external_blog_url, length: { maximum: 8000 }
 
+  def follow?(user)
+    active_relationships.exists?(followed_id: user.id)
+  end
+
+  def follow!(user_id)
+    active_relationships.create!(followed_id: user_id)
+  end
```

Comment:
引数がuserのほうが `user.follow(other_user)` みたいに呼べるのでキレイ
```suggestion
  def follow!(user)
    active_relationships.create!(followed_id: user.id)
  end
```

Reply:
user_idだとコントローラの中で、userインスタンスをとってくる必要がないのでidを受け取ろうと思ったのですが、確かに「userがfollow、他の人のidを」よりも「userがfollow、他の人を」の方が自然ですね。